boot/bootutil/src/loader.c

```diff
@@ -711,6 +719,26 @@ boot_validated_swap_type(struct boot_loader_state *state,
         } else if (rc != 0) {
             swap_type = BOOT_SWAP_TYPE_FAIL;
         }
+#ifdef CONFIG_SOC_NRF5340_CPUAPP
+        /* If the update is valid, and it targets the network core: perform the
+         * update and indicate to the caller of this function that no update is
+         * available
+         */
+        else if (rc == 0 && reset_addr > PM_CPUNET_B0N_ADDRESS) {
+            uint32_t fw_size = hdr->ih_img_size;
+
+            BOOT_LOG_INF("Starting network core update");
+            rc = do_network_core_update(vtable, fw_size);
```

Comment:
IIUC `vtable` can theoretically be 0 here. Is that ok?

Reply:
We should add some assertrs, if you look at the logic behing the check for `MAGIC` and the validation of the update, it will always be set.